shogun-manager/src/main/java/de/terrestris/shogun/manager/ShogunGsInterceptorManager.java

```diff
@@ -0,0 +1,205 @@
+import de.terrestris.shogun.interceptor.model.InterceptorRule;
+import de.terrestris.shogun.lib.enumeration.RuleType;
+import lombok.extern.log4j.Log4j2;
+import org.apache.http.client.methods.HttpDelete;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.List;
+
+/**
+ * The SHOGun-GeoServer-Interceptor manager
+ */
+@Log4j2
+public class ShogunGsInterceptorManager extends AbstractShogunManager {
+
+    final String basePathInterceptorRule = "/interceptorrules";
+
+    public ShogunGsInterceptorManager(String adminUser, String adminPassword, String shogunBaseUrl) {
+        super(adminUser, adminPassword, shogunBaseUrl);
+    }
+
+    /**
+     * Return list of all available interceptor rules from SHOGun gs-interceptor
+     * @return List of all {@link InterceptorRule}s
+     * @throws Exception The exception
+     */
+    public List<InterceptorRule> getAllInterceptorRules() throws Exception {
+        HttpGet getRequest = new HttpGet(new URI(String.format("%s%s", this.shogunServiceBaseUrl, basePathInterceptorRule)));
+        byte[] resultBytes = performRequest(getRequest);
+        if (resultBytes != null) {
+            return objectMapper.readValue(resultBytes, new TypeReference<>() {});
+        } else {
+            throw new IOException("Could not read interceptor rules from backend");
+        }
+    }
+
+    /**
+     * Return list of interceptor rules from SHOGun gs-interceptor matching passed {@link de.terrestris.shogun.interceptor.enumeration.OgcEnum.ServiceType}
+     * and {@link de.terrestris.shogun.interceptor.enumeration.HttpEnum.EventType}
+     * @param service The {@link de.terrestris.shogun.interceptor.enumeration.OgcEnum.ServiceType} to search for
+     * @param event The {@link de.terrestris.shogun.interceptor.enumeration.HttpEnum.EventType} to search for
+     * @return List of matching {@link InterceptorRule}s
+     * @throws Exception The exception
+     */
+    public List<InterceptorRule> findAllRulesForServiceAndEvent(OgcEnum.ServiceType service, HttpEnum.EventType event) throws Exception {
+        String urlString = String.format("%s%s/service/%s/event/%s", this.shogunServiceBaseUrl, basePathInterceptorRule, service, event);
+        HttpGet getRequest = new HttpGet(new URI(urlString));
+        byte[] resultBytes = performRequest(getRequest);
+        if (resultBytes != null) {
+            return objectMapper.readValue(resultBytes, new TypeReference<>() {});
+        } else {
+            throw new IOException("Could not read matching interceptor rules from backend");
+        }
+    }
+
+    /**
+     * Remove all interceptor rules for given endpoint
+     * @param endpoint The endpoint
+     * @return true if successful, false otherwise
+     * @throws Exception The exception
+     */
+    public boolean removeAllRulesForEndpoint(String endpoint) throws Exception {
```

Comment:
Absolutely minor, but I prefer that delete methods should not have a return value.